src/source-control/GitSourceControl.ts

```diff
@@ -290,39 +285,43 @@ export class GitSourceControl implements SourceControl {
       });
       queue.push(...baseCommitSttack.childCommits);
 
-      // Update refs in our hashMap
-      this.repo.commits = produce(this.repo.commits, (draftCommitHashMap) => {
-        const childrenOfTargetCommit = targetCommitSttack.childCommits;
+      // // Update refs in our hashMap
+      this.repo = produce(this.repo, (draftRepo) => {
+        const draftTargetCommitSttack = draftRepo.commits.get(targetCommit!)!;
+        const childrenOfTargetCommit = draftTargetCommitSttack.childCommits;
         const newCommitToInsert: Commit = {
           hash: newCommitOid.tostrS(),
-          title: "Sttack Commit",
+          title: baseCommitSttack.title,
           timestamp: new Date(),
```

Comment:
Is this incorrect? Didn't you just change the names here?

Reply:
Nope I didn't touch this part. Basically the commit's timestamp is obtained from the original Git commit's author/committer date (can't remember which). Since our cherry pick reuses the original commit's author and committer, I don't think either timestamp will change. But we should verify this

Reply:
Yup yup, sorry, I typoed correct as incorrect. Major slight but yes. Seems okay